source/software/module_system_basics.rst

```diff
@@ -0,0 +1,327 @@
+
+::
+
+   $ module spider Python/3.11.3-GCCcore-12.3.0
+
+   ----------------------------------------------------------------------------
+   Python: Python/3.11.3-GCCcore-12.3.0
+   ----------------------------------------------------------------------------
+    Description:
+      Python is a programming language that lets you work more quickly and
+      integrate your systems more effectively.
+
+      ...
+
+      More information
+      ================
+       - Homepage: https://python.org/
+
+      Included extensions
+      ===================
+      flit_core-3.9.0, packaging-23.1, pip-23.1.2, setuptools-67.7.2,
+      setuptools_scm-7.1.0, tomli-2.0.1, typing_extensions-4.6.3, wheel-0.40.0
+
+
+More technical information can be obtained using the ``show`` sub-command.
+It will show which other modules will be loaded and what environment paths
+will be set, e.g.:
+
+::
+
+   $ module show Python/3.11.3-GCCcore-12.3.0
+
+
+Loading modules
+~~~~~~~~~~~~~~~
+
+A module is loaded using the ``module load`` command, e.g.:
+
+::
+
+   $ module load CP2K
+
+will load the default ``CP2K`` module (``CP2K/2023.1-foss-2023a`` in this
+example).
+
+If multiple versions are installed; the ``module load`` command will
+automatically choose the default version, which is typically, but not always,
+the most recent version. If, in this example, you would prefer to use same
+version of CP2K but built with the ``intel-2023a`` toolchain, you would need
+to specify:
+
+::
+
+   $ module load CP2K/2023.1-intel-2023a
+
+.. note::
+
+   Loading modules with explicit versions is considered as best practice. It ensures
+   that your scripts will use the expected version of the software, regardless of
+   newly installed software. Failing to do this may jeopardize the reproducibility
+   of your results!
+
+Modules need not be loaded one by one; two ``load`` sub-commands
+can for example be combined as follows:
+
+::
+
+   $ module load CP2K/2023.1-foss-2023a GROMACS/2023.3-foss-2023a-PLUMED-2.9.0
+
+.. warning::
+
+   Do *not* load modules in your ``.bashrc``, ``.bash_profile`` or ``.profile``,
+   you *will* shoot yourself in the foot at some point.  Consider using
+   :ref:`module collections <collections of modules>` ``restore`` as a command
+   line alternative (so *not* in the shell initialization files either!).
+
+
+Conflicting modules
+~~~~~~~~~~~~~~~~~~~
+
+It is important to note that only modules that are compatible with
+each other should be loaded together. The loaded modules should all
+be associated with either the same toolchain or compatible (sub)toolchains
+(see also https://docs.easybuild.io/common-toolchains/#toolchains_diagram).
+
+For example, once you have loaded one or more modules that use the
+``foss/2023a`` toolchain, all other modules that you load should have been
+installed with the same toolchain or with compatible (sub)toolchains such as
+``GCC/12.3.0`` or ``GCCcore/12.3.0``.
```

Comment:
Recently, EasyBuild-based installations come more with `gompi` than `foss` and also more often with `iimpi` than with the `intel` toolchain. I recommend to mention these names here too, so that users know that `gompi` is from the same compiler family as `foss` (subset). The same for `iimpi`.

Reply:
(addressed in 4b81daa2a24f454d351bb419445960f7b5c07824)